react/features/rtcstats/RTCStats.ts

```diff
@@ -77,17 +101,21 @@ class RTCStats {
             useLegacy
         };
 
-        const rtcstatsOptions = {
-            connectionFilter,
-            pollInterval,
-            useLegacy,
-            sendSdp,
-            eventCallback: this.handleRtcstatsEvent.bind(this)
-        };
-
         this.trace = traceInit(traceOptions);
-        rtcstatsInit(this.trace, rtcstatsOptions);
-        this.initialized = true;
+        if (!this.isPeerConnectionWrapped) {
+            // FIXME we cannot unwrap the peer connection, so this has to be done once.
+            const rtcstatsOptions = {
+                connectionFilter,
+                pollInterval,
+                useLegacy,
+                sendSdp,
+                eventCallback: this.handleRtcstatsEvent.bind(this)
```

Comment:
Trying to wrap my head around this, so I may be asking dumb questions :)

In the case of mobile, where the RTCStats instance is reused, we'd be calling `maybeInit` with new options before each new conference, but the code under `if(!this.isPeerConnectionWrapped)` will only run once? Wouldn't the wrapped PC maintain a reference to the initial value of `this.trace` even after re-init?

Reply:
🤦 you're absolutely right to be suspicious of this, it can't possible work, not as it is at least.

Reply:
Just an idea, not entirely sure if would work, but I think maybe introducing a setter for the trace function is the easiest approach to updating what the wrapper is calling as its trace function.
I haven't really looked at what one would have to do for the other parameters.

Reply:
This is a very good suggestion, I will give it a try!